cg/meta/workflow/microsalt.py

```diff
@@ -306,3 +356,98 @@ def get_case_id_from_case(self, unique_id: str) -> Tuple[str, None]:
             raise click.Abort
         case_id = case_obj.internal_id
         return case_id, None
+
+    def microsalt_qc(self, case_id: str, run_dir_path: Path, lims_project: str) -> bool:
+        """Check if given microSALT case passes QC check."""
+        samples: List[Sample] = self.get_samples(case_id=case_id)
+        failed_samples: Dict = {}
+        case_qc: Dict = read_json(file_path=Path(run_dir_path, f"{lims_project}.json"))
+
+        for sample in samples:
+            if not sample.sequenced_at:
+                continue
+            self.qc_sample_check(
+                sample=sample, failed_samples=failed_samples, sample_qc=case_qc[sample.internal_id]
+            )
+
+        return self.qc_case_check(
+            case_id=case_id,
+            failed_samples=failed_samples,
+            samples=samples,
+            run_dir_path=run_dir_path,
+        )
+
+    def qc_case_check(
+        self, case_id: str, samples: List[Sample], failed_samples: Dict, run_dir_path: Path
+    ) -> bool:
+        """Perform the final QC check for a microbial case based on failed samples."""
+        qc_pass: bool = True
+
+        for sample in failed_samples:
+            # Check if negative control failed
+            if sample.control == ControlEnum.negative:
```

Comment:
```suggestion
            if sample.control != ControlEnum.negative:
```
isn't this the logic we want here?

Reply:
The logic is to first check if this is an negative control sample, then we perform another check than "regular" samples

Reply:
I see. Let's say we have failed samples, what is the scenario where we do not fail the case?

Reply:
For negative control we check if the reads are above a certain threshold. And for "regular" samples we check if the reads meets the guaranteed reads according to the apptag and if it passes 10X coverage. 

A case fails if a negative control fails, regardless of apptag. And if it is a research case than 10 % of samples are allowed to fail. If it is MWR, a case will fail as soon as one sample or more fails.

Dont know if this answers your question 😅

Reply:
I see, I found these three if-statements a bit confusing but I get it now 👍

Reply:
I understand that 😅 its a bit of a mouthful function; maybe I should add a better description or comments?

Reply:
I think it's just the logic where we allow some non-control samples to fail for MWX that makes it hard to code in an understandable way. I don't think you need to spend more time on it